swe_roe/ex1.c

```diff
@@ -116,9 +117,31 @@ PetscErrorCode Add_Buildings(User user)
 
     DMDAVecGetArrayDOF  (da, user->B, &b_ptr);
     VecZeroEntries(user->B);
+    /*
+
+    x represents the reflecting wall,
+    | represents the dam that will be broken suddenly.
+    hu is the upsatream water dpeth, and hd is the downstream water depth.
+
+    x x x x x x x x x x x
+    x         x         x
+    x         x         x
+    x         |         x
+    x         |         x
+    x    hu   |    hd   x
+    x         x         x
+    x         x         x
+    x         x         x
+    x         x         x
+    x x x x x x x x x x x
+    
+    */
     for (j = user->ys; j < user->ys + user->ym; j = j + 1) {
         for (i = user->xs; i < user->xs + user->xm; i = i + 1) {
-            if (i > xmin && i < xmax && j > ymin && j < ymax) {
+            if (i < 30 && j >= 95 && j < 105) {
+                b_ptr[j][i][0] = 1.;
+            }
+            else if (i >= 105 && j >= 95 && j < 105) {
```

Comment:
I believe the location of the wall is assuming `user->hx = user->hy = 1.0`. The code should be modified to explicitly take into account `hx` and `hy`, so the wall remains in the same location when the grid spacing is modified.

Reply:
@donghuix Let me know if you think addressing this comment requires significant work that you would prefer to fix in a future PR.